modules/claims_api/app/controllers/claims_api/v2/veterans/intent_to_file_controller.rb

```diff
@@ -8,12 +8,18 @@ module Veterans
       class IntentToFileController < ClaimsApi::V2::ApplicationController
         before_action :verify_access!
 
-        ITF_TYPES = %w[compensation pension burial].freeze
-
+        # GET to fetch active intent to file by type
+        #
+        # @return [JSON] ITF record
         def type
+          itf_types = ClaimsApi::V2::ParamsValidation::IntentToFile::IntentToFileValidator.itf_types
+          unless itf_types.include?(request.path_parameters[:type])
+            message = "Invalid type parameter: '#{params[:type]}'"
```

Comment:
why are we using `request.path_parameters` here? and then we use `params` right below?